_posts/2023-11-01-#28368.md

```diff
@@ -7,18 +7,79 @@ authors: [ismaelsadeeq]
+
+
+* This PR removes the [`CTxMempool`](https://github.com/bitcoin/bitcoin/blob/22fa1f4702e9a60e141f545f8d09704deca34b22/src/txmempool.h#L301) dependency on the [`CBlockPolicyEstimator`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/policy/fees.h#L146C13-L146C13) by making `CBlockPolicyEstimator` a client of [`CValidationInterface`](https://github.com/bitcoin/bitcoin/blob/9c30f5ef9d6712539cb8dc0f52b3a2206a25e074/src/validationinterface.h#L77), instead of an internal component of `CTxMemPool`.
+
+* The `CBlockPolicyEstimator` now updates asynchronously in response to [`CValidationInterface`](https://github.com/bitcoin/bitx`coin/blob/9c30f5ef9d6712539cb8dc0f52b3a2206a25e074/src/validationinterface.h#L77) events.
+
+* The PR added another [`CValidationInterface`](https://github.com/bitcoin/bitcoin/blob/9c30f5ef9d6712539cb8dc0f52b3a2206a25e074/src/validationinterface.h#L77) event `MempoolTransactionsRemovedForConnectedBlock` which is fired in [`removeForBlock`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/txmempool.h#L487).
+
+* The PR also modifies the [`TransactionAddedToMempool`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/validationinterface.h#L99) event parameter from `CTransactionRef` to a new struct which has the [`CTransactionRef`](https://github.com/bitcoin/bitcoin/blob/ab2f531b785c3c17746ac9766a6db5b6d26677e8/src/primitives/transaction.h#L421) and other information the fee estimator needs such as the size, fees, and others. There has been an attempt at this PR [#11775](https://github.com/bitcoin/bitcoin/pull/11775)
+
 ## Questions
+
+### Conceptual
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
 
+2. Why is it beneficial to remove `CTxMempool` dependency on `CBlockPolicyEstimator`?
+
+3. Are there any benefits of the `CBlockPolicyEstimator` being a member of `CBlockPolicyEstimator` Are there downsides to removing it?
+
+4. The first attempt [#11775](https://github.com/bitcoin/bitcoin/pull/11775) split the `CValidationInterface` into `CValidationInterface` and `MempoolInterface`. What is the distinction between the two interfaces?
+
+5. Is the split necessary to achieve a background fee estimator? Is it crucial in the long run?
+
+6. Why does this approach omit the split?
+
+7. Can you list the other subscribers to the interface? (Hint: look for classes that inherit from `CValidationInterface`)
+
+8. What does it mean for a class to implement one of the `CValidationInterface` methods? Why is it similar to "subscribing to the event"?
+
+9. What is the difference between implementing a `BlockConnected` and a `NewPoWValidBlock` callback, apart from the fact that they are different events? On which thread is each callback executed?
```

Comment:
I think you're trying to ask about (a)synchronousness? I would phrase it in specifically that way. The difference between implementing a `BlockConnected` and a `NewPoWValidBlock` callback is very broad.

Reply:
Updated, thank you